### PR TITLE
Simplify the Async USB plugin

### PR DESCRIPTION
**Pull request by nomis52**

### CODE REVIEW

**Comment by nomis52** [comment]
Blocked behind #929

File: plugins/usbdmx/AsyncPluginImpl.cpp [hunk 1]
```diff
@@ -253,22 +268,17 @@ void AsyncPluginImpl::USBDeviceAdded(libusb_device *usb_device) {
...
 
 /*
- * @brief Signal widget / device addition.
- * @param device_id The device id to add.
+ * @brief Called when a new OLA device is ready.
+ * @param widget The Widget used for this device.
  * @param device The new olad device that uses this new widget.
  *
  * This is run within the main thread.
 # */
```
Comment (peternewman) [inline plugins/usbdmx/AsyncPluginImpl.cpp]: Remove the rest#

Comment (nomis52) [inline plugins/usbdmx/AsyncPluginImpl.cpp]: Done

File: plugins/usbdmx/AnymauDMXFactory.cpp [hunk 2]
```diff
@@ -42,8 +42,7 @@ bool AnymauDMXFactory::DeviceAdded(
     WidgetObserver *observer,
     libusb_device *usb_device,
```
Comment (peternewman) [inline plugins/usbdmx/AnymauDMXFactory.cpp]: Remove this?

File: plugins/usbdmx/AnymauDMXFactory.cpp [hunk 3]
```diff
@@ -42,8 +42,7 @@ bool AnymauDMXFactory::DeviceAdded(
...
     const struct libusb_device_descriptor &descriptor) {
-  if (descriptor.idVendor != VENDOR_ID || descriptor.idProduct != PRODUCT_ID ||
-      HasDevice(usb_device)) {
```
Comment (peternewman) [inline plugins/usbdmx/AnymauDMXFactory.cpp]: How come we've dropped the test for this?

File: plugins/usbdmx/AsyncPluginImpl.cpp [hunk 4]
```diff
@@ -196,16 +203,28 @@ void AsyncPluginImpl::WidgetRemoved(VellemanK8062 *widget) {
...
-      OLA_INFO << "USB device " << device_id << " removed";
-      state->factory->DeviceRemoved(&m_widget_observer, state->usb_device);
-      state->factory = NULL;
+    DeviceState *state;
+    if (!STLLookupAndRemove(&m_device_map, device_id, &state) || !state) {
+      return;
     }
-    STLRemoveAndDelete(&m_device_map, device_id);
+
+    // At some point we'll need to notify the factory here. For instance in the
```
Comment (peternewman) [inline plugins/usbdmx/AsyncPluginImpl.cpp]: Make this a TODO

**Comment by peternewman** [comment]
Generally LGTM

File: plugins/usbdmx/AnymauDMXFactory.cpp (hunk 3 above)
Comment (nomis52) [inline plugins/usbdmx/AnymauDMXFactory.cpp]: De-duping is done in the hotplug agent now.

File: plugins/usbdmx/AsyncPluginImpl.cpp (hunk 4 above)
Comment (nomis52) [inline plugins/usbdmx/AsyncPluginImpl.cpp]: There isn't an actionable item. It's more of a 'someone will need to add code here sometime in the future'

File: plugins/usbdmx/AnymauDMXFactory.cpp (hunk 3 above)
Comment (peternewman) [inline plugins/usbdmx/AnymauDMXFactory.cpp]: Ah okay

File: plugins/usbdmx/AsyncPluginImpl.cpp (hunk 4 above)
Comment (peternewman) [inline plugins/usbdmx/AsyncPluginImpl.cpp]: When we add more functionality elsewhere?

Comment (nomis52) [inline plugins/usbdmx/AsyncPluginImpl.cpp]: Yeah, if someone wants to make the f/w loading async, they'll need to add a call in here.

Comment (peternewman) [inline plugins/usbdmx/AsyncPluginImpl.cpp]: Understood. So does Sunlite only work in sync mode?

**Comment by peternewman** [comment]
LGTM

File: plugins/usbdmx/AsyncPluginImpl.cpp (hunk 4 above)
Comment (nomis52) [inline plugins/usbdmx/AsyncPluginImpl.cpp]: It works in both, but the f/w loading is always synchronous.

Comment (peternewman) [inline plugins/usbdmx/AsyncPluginImpl.cpp]: Okay

**Pull request merged**